src/app/public/modules/action-button/action-button.component.html

```diff
@@ -2,47 +2,84 @@
   class="sky-action-button sky-btn-default sky-rounded-corners"
   tabindex="0"
   [fragment]="permalink.route?.extras?.fragment"
+  [skyThemeClass]="{
+    'sky-btn sky-box' : 'modern'
+  }"
   [queryParams]="permalink.route?.extras?.queryParams"
   [queryParamsHandling]="permalink.route?.extras?.queryParamsHandling"
   [routerLink]="permalink.route?.commands"
 >
-  <ng-container *ngTemplateOutlet="actionButtonContent">
+  <ng-container *ngTemplateOutlet="
+    themeName === 'modern' ? modernContent : defaultContent
+  ">
   </ng-container>
 </a>
 
 <a *ngIf="permalink && !permalink.route"
   class="sky-action-button sky-btn-default sky-rounded-corners"
   tabindex="0"
-  [href]="permalink.url">
-  <ng-container *ngTemplateOutlet="actionButtonContent">
+  [href]="permalink.url"
+  [skyThemeClass]="{
+    'sky-btn sky-box' : 'modern'
+  }"
+>
+  <ng-container *ngTemplateOutlet="
+    themeName === 'modern' ? modernContent : defaultContent
+  ">
   </ng-container>
 </a>
 
 <div *ngIf="!permalink"
   class="sky-action-button sky-btn-default sky-rounded-corners"
   tabindex="0"
   role="button"
+  [skyThemeClass]="{
+    'sky-btn sky-box' : 'modern'
+  }"
   (click)="buttonClicked()"
   (keydown.enter)="enterPress()"
 >
-  <ng-container *ngTemplateOutlet="actionButtonContent">
+  <ng-container *ngTemplateOutlet="
+    themeName === 'modern' ? modernContent : defaultContent
```

Comment:
Would it be better to use the new `*skyThemeIf` structural directive? Curious your thoughts.
```
<!-- Revert to using this guy again: -->
<ng-container *ngTemplateOutlet="actionButtonContent">
</ng-container>

<!-- Then create separate containers within the template, for the logic: -->
<ng-template #actionButtonContent>
  <ng-container *skyThemeIf="'default'">
    ...
  </ng-container>
  <ng-container *skyThemeIf="'modern'">
    ...
  </ng-container>
</ng-template>
```

Reply:
I like that. It cleans up the `.ts` file as well. Done.